src/shared/Microsoft.Git.CredentialManager/Authentication/MicrosoftAuthentication.cs

```diff
@@ -360,6 +398,33 @@ public HttpClient GetHttpClient()
 
         #region Auth flow capability detection
 
+        private bool CanUseBroker()
+        {
+#if NETFRAMEWORK
+            // We only support the broker on Windows 10 and require an interactive session
+            if (!Context.SessionManager.IsDesktopSession || !PlatformUtils.IsWindows10())
```

Comment:
There is a helper for this in MSAL that we've added in collaboration with you: 

https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/2cebdaec10f1991482ed7109a2fa34389a037299/src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs#L318

Reply:
Yeah, looking at this however we won't be able to use this sadly.. it's hanging off of the PCABuilder, and we need it before and after the builder exists 😢